src/Component/SignUp.js

```diff
@@ -238,19 +330,27 @@ const SignUp = () => {
       timer = setTimeout(() => callback(...args), delay);
     };
   };
-  const printValue = useCallback(
-    debounce(() => onClickIdCheck(), 3000),
+  const printIdValue = useCallback(
+    debounce(() => onClickIdCheck(), 1000),
+    []
+  );
+  const printPhoneNumValue = useCallback(
+    debounce(() => onClickPhoneNumCheck(), 1000),
+    []
+  );
+  const printEmailValue = useCallback(
+    debounce(() => onClickEmailCheck(), 1000),
     []
   );
 
   const onChangeId = (e) => {
-    printValue(e.target.value);
     setInputValue({
       ...inputValue,
-      [e.target.name]: e.target.value,
+      [e.target.name]: idInput.current.value,
     });
     if (idInput.current.value.length >= 8) {
-      setIdValid(true);
+      // 8글자 넘고 중복검사중에 변화 안되게 바꿔야함
+      printIdValue(idInput.current.value);
```

Comment:
```suggestion
      debouncedOnClickIdCheck(idInput.current.value);
```
print 만 보고 무엇을 하는지 파악하는데 시간이 걸렸습니다.

이해하기 쉽게 debounced 표현을 쓰면 어떨까요?

Reply:
예